contracts/Registry.vy

```diff
@@ -0,0 +1,254 @@
+# @version 0.2.8
+
+interface DetailedERC20:
+    def name() -> String[52]: view
+    def symbol() -> String[30]: view
+
+
+interface Vault:
+    def token() -> address: view
+    def apiVersion() -> String[28]: view
+    def governance() -> address: view
+    def initialize(
+        token: address,
+        governance: address,
+        rewards: address,
+        name: String[64],
+        symbol: String[32],
+        guardian: address,
+    ): nonpayable
+
+
+# len(Vault.releases)
+nextRelease: public(uint256)
+releases: public(HashMap[uint256, address])
+
+# Token.address => len(Vault.deployments)
+nextDeployment: public(HashMap[address, uint256])
+vaults: public(HashMap[address, HashMap[uint256, address]])
+
+# 2-phase commit
+governance: public(address)
+pendingGovernance: address
+
+tags: public(HashMap[address, String[1000000]])
+banksy: public(HashMap[address, bool])  # could be anyone
+
+event NewRelease:
```

Comment:
per @fubuloubu \` list of Vault releases, and an event stream to track them (just like with deployments)\`